src/qz/ui/IconCache.java

```diff
@@ -231,15 +236,73 @@ public void setBgColor(Icon i, Color bgColor) {
     }
 
     /**
-     * Inverts the colors for a particular icon
-     * For integration with dark or light desktop themes
-     *
-     * @param i the IconCache.Icon
+     * Replaces the cached tray icons with corrected versions if necessary
+     * e.g.
+     *  - Ubuntu transparency
+     *  - macOS masked icons
+     *  - macOS 10.14+ dark mode support
      */
-    public void invertColors(Icon i) {
+    public void fixTrayIcons(boolean darkMode) {
+        // Fix the tray icon to look proper on Ubuntu
+        if (SystemUtilities.isUbuntu()) {
+            UbuntuUtilities.fixTrayIcons(this);
+        }
+
+        // Handle mask-style tray icons
+        if (SystemUtilities.prefersMaskTrayIcon()) {
+            // Clone the mask icon
+            for (String id : Icon.MASK_ICON.getIds()) {
+                BufferedImage src = images.get(id);
+                BufferedImage dest = clone(src);
```

Comment:
Why not use `images.get(id)` directly if we aren't going to use the `src` reference anywhere else?

Reply:
Done via bd4937e.